lib/commons/color/get-background-color.js

```diff
@@ -57,11 +94,81 @@ function calculateObscuringAlpha(elmIndex, elmStack) {
 	}
 	return totalAlpha;
 }
+/**
+ * Determine if element is partially overlapped, triggering a Can't Tell result
+ * @private
+ * @param {Element} elm
+ * @param {Element} bgElm
+ * @param {Object} bgColor
+ * @return {Boolean}
+ */
+function elmPartiallyObscured(elm, bgElm, bgColor) {
+	var obscured = (elm !== bgElm && !dom.visuallyContains(elm, bgElm) && bgColor.alpha !== 0);
+	if (obscured) {
+		axe.commons.color.incompleteData.set('bgColor', 'elmPartiallyObscured');
+	}
+	return obscured;
+}
 
-function elmOutsideParent(elm, bgElm, bgColor) {
-	var visible = (elm !== bgElm && !dom.visuallyContains(elm, bgElm) && bgColor.alpha !== 0);
-	return visible;
+/**
+ * Include nodes missing from initial gathering because
+ * document.elementsFromPoint misses some elements we need
+ * i.e. TR is missing from table elementStack and leaves out bgColor
+ * https://github.com/dequelabs/axe-core/issues/273
+ *
+ * @private
+ * @param {Array} elmStack
+ * @param {Element} elm
+ */
+function includeMissingElements(elmStack, elm) {
```

Comment:
This function relies on mutation. It's just as easy to have this function return a new array and not mutate the elmStack. Generally, we shouldn't rely on mutation if we can avoid it. Pure functions are easier to test and use.